### PR TITLE
boards: Add BT510 support

### PR DESCRIPTION
**Pull request by greg-leach**
Add support for the Laird Connectivity BT510 sensor

Signed-off-by: Greg Leach <greg.leach@lairdconnect.com>

### CODE REVIEW

File: tests/kernel/threads/no-multithreading/boards/bt510.conf [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (andrewboie) [inline tests/kernel/threads/no-multithreading/boards/bt510.conf:2]: I don't see any changes to testcase.yaml to use this configuration or the overlay file. Need to wire this to CI.

Comment (greg-leach) [reply tests/kernel/threads/no-multithreading/boards/bt510.conf:2]: This test case will fail without this overlay - I2C drivers do not support single threaded operation.

**Review by rerickson1 — APPROVED** [review]

File: tests/kernel/threads/no-multithreading/boards/bt510.conf [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (rerickson1) [inline tests/kernel/threads/no-multithreading/boards/bt510.conf:1]: Many settings in this are duplicated.  Only need to disable I2C.

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: tests/kernel/threads/no-multithreading/boards/bt510.conf (hunk 2 above)
Comment (greg-leach) [reply tests/kernel/threads/no-multithreading/boards/bt510.conf:1]: Updated to only disable I2C devices.

**Review by tyrantlizard77 — APPROVED** [review]

File: boards/arm/bt510/doc/bt510.rst [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+LED
+---
+
+* LED1A (green) = P0.22
+* LED1B (red) = P0.20
+
+Push button
+------------
+
+* BUTTON = SW1 = P0.42
```
Comment (thedjnK) [inline boards/arm/bt510/doc/bt510.rst:106]: Should be P1.10 (nRF52840 mapping) or SIO42 (BL654/schematic mapping)

File: boards/arm/bt510/doc/bt510.rst [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+* RESET = SW2 = nReset/IF BOOT
```
Comment (thedjnK) [inline boards/arm/bt510/doc/bt510.rst:107]: No such button on board

File: boards/arm/bt510/doc/bt510.rst [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: bt510
+   :goals: build flash
+
+Debugging
+=========
+
+Refer to the :ref:`nordic_segger` page to learn about debugging Nordic based boards with a
+Segger IC.
```
Comment (thedjnK) [inline boards/arm/bt510/doc/bt510.rst:186]: It requires an external programmer, it does not have an on-board one

File: boards/arm/bt510/doc/bt510.rst [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+* :abbr:`I2C (Inter-Integrated Circuit)`
+* :abbr:`MPU (Memory Protection Unit)`
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+* :abbr:`PWM (Pulse Width Modulation)`
+* RADIO (Bluetooth Low Energy and 802.15.4)
+* :abbr:`RTC (nRF RTC System Clock)`
+* Segger RTT (RTT Console)
+* :abbr:`SPI (Serial Peripheral Interface)`
+* :abbr:`UART (Universal Asynchronous Receiver-Transmitter)`
+* :abbr:`USB (Universal Serial Bus)`
```
[the remark this answers is not in the document]
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:26]: Updated.

File: boards/arm/bt510/doc/bt510.rst [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+* FLASH
+* :abbr:`GPIO (General Purpose Input Output)`
+* :abbr:`I2C (Inter-Integrated Circuit)`
+* :abbr:`MPU (Memory Protection Unit)`
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+* :abbr:`PWM (Pulse Width Modulation)`
+* RADIO (Bluetooth Low Energy and 802.15.4)
+* :abbr:`RTC (nRF RTC System Clock)`
+* Segger RTT (RTT Console)
+* :abbr:`SPI (Serial Peripheral Interface)`
```
[the remark this answers is not in the document]
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:24]: Updated.

File: boards/arm/bt510/doc/bt510.rst [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
++-----------+------------+----------------------+
+| RTC       | on-chip    | system clock         |
++-----------+------------+----------------------+
+| RTT       | Segger     | console              |
++-----------+------------+----------------------+
+| SPI(M/S)  | on-chip    | spi                  |
++-----------+------------+----------------------+
+| UART      | on-chip    | serial               |
++-----------+------------+----------------------+
+| USB       | on-chip    | usb                  |
```
[the remark this answers is not in the document]
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:85]: Updated.

File: boards/arm/bt510/doc/bt510.rst (hunk 3 above)
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:106]: Updated.

File: boards/arm/bt510/doc/bt510.rst (hunk 4 above)
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:107]: Updated.

File: boards/arm/bt510/doc/bt510.rst (hunk 5 above)
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:186]: Updated.

File: boards/arm/bt510/doc/bt510.rst [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+NOTE: On the BT510, the FTDI USB should be used to access the UART console.
+
+.. code-block:: console
+
+   $ minicom -D <tty_device> -b 115200
+
+Replace :code:`<tty_device>` with the port where the bt510 can be found. For example, under Linux, :code:`/dev/ttyUSB0`.
+
+Then build and flash the application in the usual way.
```
Comment (rerickson1) [inline boards/arm/bt510/doc/bt510.rst:168]: Maybe change this to mention you cannot flash any firmware unless you have an external debugger.

File: boards/arm/bt510/doc/bt510.rst [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+and configure all the necessary software. Further information can be
+found in :ref:`nordic_segger_flashing`. Then build and flash
+applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Here is an example for the :ref:`hello_world` application.
+
+First, run your favorite terminal program to listen for output.
+
+NOTE: On the BT510, the FTDI USB should be used to access the UART console.
```
Comment (rerickson1) [inline boards/arm/bt510/doc/bt510.rst:160]: There is no FTDI USB on the BT510 itself.  Mention the adapter board needs to be used to connect the debug UART.

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: boards/arm/bt510/doc/bt510.rst (hunk 9 above)
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst:170]: Updated.

File: boards/arm/bt510/doc/bt510.rst (hunk 10 above)
Comment (greg-leach) [reply boards/arm/bt510/doc/bt510.rst]: Updated.

File: boards/arm/bt510/bt510.dts [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+	model = "Laird Sentrius BT510 Sensor";
+	compatible = "nordic,pca10056-dk";
```
Comment (rerickson1) [inline boards/arm/bt510/bt510.dts:12]: Our board is not technically compatible with the `nrf52840dk_nrf52840` (PCA10056).  Change this to `lairdconnect,bt510`

**Review by rerickson1 — CHANGES_REQUESTED** [review]

File: boards/arm/bt510/bt510.dts (hunk 11 above)
Comment (greg-leach) [reply boards/arm/bt510/bt510.dts]: Fixed.

**Review by rerickson1 — APPROVED** [review]

File: boards/arm/bt510/bt510_defconfig [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+# enable uart driver
+CONFIG_SERIAL=y
+
+# enable accelerometer & temperature sensor
+CONFIG_I2C=y
+CONFIG_SENSOR=y
+CONFIG_LIS2DH=y
+CONFIG_LIS2DH_TRIGGER_GLOBAL_THREAD=y
+CONFIG_LIS2DH_OPER_MODE_LOW_POWER=y
+CONFIG_SI7055=y
```
Comment (MaureenHelm) [inline boards/arm/bt510/bt510_defconfig:25]: Are you sure you want to enable these here? They will be enabled by default for all applications, including hello_world and blinky.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Can you reduce the size of the images?

File: boards/arm/bt510/bt510_defconfig (hunk 12 above)
Comment (greg-leach) [reply boards/arm/bt510/bt510_defconfig:25]: PNG replaced with JPG to reduce physical size, both images reduced down to 250px in RST file.

The I2C sensors need to be enabled by default. I've created project overlay files to disable them for single threaded applications due to the underlying I2C drivers spawining additional threads.

**Review by carlescufi — APPROVED** [review]
Looks good, thank you!

**Comment by carlescufi** [comment]
@MaureenHelm this should be ready now

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**